### PR TITLE
fix(IDP-134): fix table border and default pagination

### LINKED ISSUE #291: Allow Table pagination to keep default values if not passed explicitly
## The feature or bug you are proposing

In order to keep the default pagination the table now requires to pass 
```pagination={...Table.pagination, ...customPartialPagination}```

while it would be better to just pass
```pagination={customPartialPagination}```

## The description of the bug or the rationale of your proposal

Easier developer experience in using the Table component

### PR DESCRIPTION
**Pull request by Giogia**
### Description

Fix Table style and default pagination

fixes #291

##### Table

Added lateral border
Keep pagination default values if not overwritten

### Issue ticket number and link

[IDP-134](https://makeitapp.atlassian.net/browse/IDP-134)

### Checklist

- [x] added the link to the Jira task
- [x] commit message and branch name follow conventions
- [x] tests are included
- [x] changes are accessible and documented from components stories
- [x] typings are updated or integrated accordingly with your changes
- [ ] all added components are exported from index file (if necessary)
- [ ] all added files include Apache 2.0 license
- [x] you are not committing extraneous files or sensible data
- [ ] the browser console does not have any logged errors
- [x] necessary labels have been applied to this pull request (enhancement, bug, ecc.)

### CODE REVIEW

File: src/components/Table/Table.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const Table = <RecordType extends GenericRecord>({
         footer={footer}
         loading={false}
         locale={intlLocale}
-        pagination={pagination}
+        pagination={{ ...Table.pagination, ...pagination }}
```
Comment (Danielecina) [inline src/components/Table/Table.tsx:68]: Serve davvero? Nelle default props tanto le passiamo, se l'utente decide che non le vuole glielo lascerei fare?

**Review by RiccardoCorona-MiaPlatform — DISMISSED** [review]

File: src/components/Table/Table.tsx (hunk 1 above)
Comment (fredmaggiowski) [inline src/components/Table/Table.tsx:68]: Could a `useMemo` be helpful?

Comment (Giogia) [reply src/components/Table/Table.tsx:68]: This is the fix requested in #291 . 

In this way, we allow to provide only a partial default pagination, overwriting only the necessary fields while maintaining the other default values. From the user's perspective, there is no need to spread Table.pagination explicitly.

Comment (Giogia) [reply src/components/Table/Table.tsx]: Added ✅

Comment (Danielecina) [reply src/components/Table/Table.tsx]: ok, I see that the code is changed 👍

**Review by Danielecina — APPROVED** [review]

**Review by RiccardoCorona-MiaPlatform — APPROVED** [review]

**Pull request merged**